### PR TITLE
fix missing return statement. Do not normalize remote paths

### PR DESCRIPTION
**Pull request by f4hy**
## What does this PR do?
Fixes two issues found with #2164 merged yesterday. Local paths worked, but remote directory writing broke in a rebase. My bad.

This was now tested by training and passing `--default_root_dir='s3://mybucket/` at training and verified all files write successfully to an `s3` bucket. `v_num` successfully increments on multiple runs. Pointing a tensorboard at the s3 bucket successfully live updates while model is training. 

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [X] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
yup!

### CODE REVIEW

File: pytorch_lightning/utilities/cloud_io.py [hunk 1]
```diff
@@ -35,10 +35,11 @@ def modern_gfile():
...
     """
     tb_version = version.parse(tensorboard.version.VERSION)
-    modern_gfile = tb_version >= version.parse('2.0')
+    modern_gfile = tb_version >= version.parse("2.0")
+    return modern_gfile
 
 
-def cloud_open(path: pathlike, mode: str, newline:str = None):
+def cloud_open(path: pathlike, mode: str, newline: str = None):
     if sys.platform == "win32":
```
Comment (rohitgr7) [inline pytorch_lightning/utilities/cloud_io.py:43]:
```suggestion
    if platform.system() == 'Windows'
```
You can `import platform` above.

**Comment by f4hy** [comment]
@rohitgr7 Thanks! Added your suggestion.

File: pytorch_lightning/utilities/cloud_io.py [hunk 2]
```diff
@@ -35,11 +35,12 @@ def modern_gfile():
...
+    return modern_gfile
 
 
-def cloud_open(path: pathlike, mode: str, newline:str = None):
-    if sys.platform == "win32":
+def cloud_open(path: pathlike, mode: str, newline: str = None):
+    if platform.system() == "Windows":
         log.debug(
             "gfile does not handle newlines correctly on windows so remote files are not"
             "supported falling back to normal local file open."
```
Comment (awaelchli) [inline pytorch_lightning/utilities/cloud_io.py:46]:
```suggestion
            "gfile does not handle newlines correctly on windows so remote files are not"
            " supported falling back to normal local file open."
```
I think this is needed or else there is no space

**Comment by f4hy** [comment]
Just updated the PR to add documentation of the remote directory support.

**Comment by awaelchli** [comment]
@f4hy The last sentence is missing a word I think

**Comment by f4hy** [comment]
> @f4hy The last sentence is missing a word I think

Hmm. Last sentence where? Sorry, not seeing it. I am sure you are right.

File: pytorch_lightning/trainer/__init__.py [hunk 3]
```diff
@@ -308,10 +308,12 @@ def on_train_end(self, trainer, pl_module):
...
-Default path for logs and weights when no logger
-or :class:`pytorch_lightning.callbacks.ModelCheckpoint` callback passed.
-On certain clusters you might want to separate where logs and checkpoints
-are stored. If you don't then use this argument for convenience.
+Default path for logs and weights when no logger or
+:class:`pytorch_lightning.callbacks.ModelCheckpoint` callback passed.  On
+certain clusters you might want to separate where logs and checkpoints are
+stored. If you don't then use this argument for convenience. Paths can be local
+paths or remote paths such as `s3://bucket/path` or 'hdfs://path/'. Credentials
+will need to be set up use remote filepaths.
```
Comment (awaelchli) [inline pytorch_lightning/trainer/__init__.py:316]:
```suggestion
paths or remote paths such as `s3://bucket/path` or 'hdfs://path/'. Credentials
will need to be set up to use remote filepaths.
```
not sure

Comment (f4hy) [reply pytorch_lightning/trainer/__init__.py]: Nice catch thanks. I read that 3 times myself and thought it was fine, but you are correct. I blame English being my first language.

**Pull request merged**